src/rtc.rs

```diff
@@ -0,0 +1,469 @@
+//! Interface to the real time clock. See STM32F303 reference manual, section 27.
+//! For more details, see
+//! [ST AN4759](https:/www.st.com%2Fresource%2Fen%2Fapplication_note%2Fdm00226326-using-the-hardware-realtime-clock-rtc-and-the-tamper-management-unit-tamp-with-stm32-microcontrollers-stmicroelectronics.pdf&usg=AOvVaw3PzvL2TfYtwS32fw-Uv37h)
+
+use crate::pac::{PWR, RTC};
+use crate::rcc::{APB1, BDCR};
+use core::convert::TryInto;
+use rtcc::{Datelike, Hours, NaiveDate, NaiveDateTime, NaiveTime, Rtcc, Timelike};
+
+/// Invalid input error
+#[derive(Debug)]
+pub enum Error {
+    InvalidInputData,
+}
+
+pub const LSE_BITS: u8 = 0b01;
+
+pub struct Rtc {
+    pub regs: RTC,
+}
+
+impl Rtc {
+    /// Create and enable a new RTC, and configure its clock source and prescalers.
+    /// From AN4759, Table 7, when using the LSE (The only clock source this module
+    /// supports currently), set `prediv_s` to 255, and `prediv_a` to 127 to get a
+    /// calendar clock of 1Hz.
+    /// The `bypass` argument is `true` if you're using an external oscillator that
+    /// doesn't connect to `OSC32_IN`, such as a MEMS resonator.
+    pub fn new(
+        regs: RTC,
+        prediv_s: u16,
+        prediv_a: u8,
+        bypass: bool,
+        apb1: &mut APB1,
+        bdcr: &mut BDCR,
```

Comment:
Taking feedback that I received on my PR, you should not accept the ABP1 and BDCR structs. Rather, you should use bit banding to enable the bits.

Reply:
I have made the requested changes, hopefully this is what was being looked for.